src/stx/stl/trex_stl_stream_node.h

```diff
@@ -703,5 +703,85 @@ friend class TrexStatelessDpPerPort;
 
 static_assert(sizeof(CGenNodePCAP) == sizeof(CGenNode), "sizeof(CGenNodePCAP) != sizeof(CGenNode)" );
 
-#endif /* __TREX_STL_STREAM_NODE_H__ */
+/* this is a event for time synchronization. */
+struct CGenNodeTimesync : public CGenNodeBase {
```

Comment:
This node is called from all the DP cores. My understanding that you should have only **one** for that.

Reply:
For TRex instances using high res ticks or system RTC one CGenNodeTimesync will be enough.  But I am working on a potentially different case.  If I get it correctly, a TRex core kind of represents DPDK port (physical port on a NIC).  If this is the case, then I would stick to this proposal.  From my understanding, some NICs have RTCs attached to ports which may be used in PTP time synchronization.